scrunch/datasets.py

```diff
@@ -1441,12 +1481,34 @@ class Variable(object):
 
     CATEGORICAL_TYPES = {'categorical', 'multiple_response', 'categorical_array'}
 
-    def __init__(self, resource, dataset_resource):
-        self.resource = resource
-        self.url = self.resource.self
-        self.dataset = Dataset(dataset_resource)
+    def __init__(self, var_shoji, dataset):
+        self.shoji_tuple = var_shoji
+        self.is_instance = False
+        self._resource = None
+        self.url = None
+        # pass an Entity class to save time
+        if isinstance(dataset, pycrunch.shoji.Entity):
```

Comment:
Do we have any case where we send the shoji entity on Variable init now? I ask so we don't unnecessarily support receiving the shoji entity and do hacks here leaving unused code.